plugins/modules/web_infrastructure/jira.py

```diff
@@ -662,35 +410,263 @@ def main():
+            raise ValueError('at least one of filename or content must be provided')
+        mime = v.attachment.get('mimetype')
+
+        if not os.path.isfile(filename):
+            raise ValueError('The provided filename does not exist: %s' % filename)
+
+        content_type, data = self._prepare_attachment(filename, content, mime)
+
+        url = v.restbase + '/issue/' + v.issue + '/attachments'
+        return True, self.post(
+            url, v.username, v.password, v.timeout, data, content_type=content_type,
+            additional_headers={"X-Atlassian-Token": "no-check"}
+        )
+
+    # Ideally we'd just use prepare_multipart from ansible.module_utils.urls, but
+    # unfortunately it does not support specifying the encoding and also defaults to
+    # base64. Jira doesn't support base64 encoded attachments (and is therefore not
+    # spec compliant. Go figure). I originally wrote this function as an almost
+    # exact copypasta of prepare_multipart, but ran into some encoding issues when
+    # using the noop encoder. Hand rolling the entire message body seemed to work
+    # out much better.
+    #
+    # https://community.atlassian.com/t5/Jira-questions/Jira-dosen-t-decode-base64-attachment-request-REST-API/qaq-p/916427
+    #
+    # content is expected to be a base64 encoded string since Ansible doesn't
+    # support passing raw bytes objects.
+    @staticmethod
+    def _prepare_attachment(filename, content=None, mime_type=None):
+        def escape_quotes(s):
+            return s.replace('"', '\\"')
+
+        boundary = "".join(random.choice(string.digits + string.ascii_letters) for dummy in range(30))
+        name = to_native(os.path.basename(filename))
+
+        if not mime_type:
+            try:
+                mime_type = mimetypes.guess_type(filename or '', strict=False)[0] or 'application/octet-stream'
+            except Exception:
+                mime_type = 'application/octet-stream'
+        main_type, sep, sub_type = mime_type.partition('/')
+
+        if not content and filename:
+            with open(to_bytes(filename, errors='surrogate_or_strict'), 'rb') as f:
+                content = f.read()
+        else:
+            try:
+                content = base64.b64decode(content)
+            except binascii.Error as e:
+                raise Exception("Unable to base64 decode file content: %s" % e)
+
+        lines = [
+            "--{0}".format(boundary),
+            'Content-Disposition: form-data; name="file"; filename={0}'.format(escape_quotes(name)),
+            "Content-Type: {0}".format("{0}/{1}".format(main_type, sub_type)),
+            '',
+            to_text(content),
+            "--{0}--".format(boundary),
+            ""
+        ]
+
+        return (
+            "multipart/form-data; boundary={0}".format(boundary),
+            "\r\n".join(lines)
+        )
+
+    def request(
+            self,
+            url,
+            user,
+            passwd,
```

Comment:
As a later refactoring, it probably makes sense to remove `user` and `passwd`, since they are always the same.

Reply:
Yup, it makes sense, removing those and `timeout` as well